src/app/[locale]/layout.tsx

```diff
@@ -0,0 +1,20 @@
+import { useLinguiSSR } from "@/i18n/i18n";
+import { LinguiProvider } from "@/i18n/lingui-provider";
+import { Trans } from "@lingui/macro";
+
+export default async function LocaleLayout({
+  params,
+  children,
+}: {
+  params: any;
+  children: React.ReactNode;
+}) {
+  const i18n = await useLinguiSSR(params.locale);
+
+  return (
+    <LinguiProvider locale={i18n.locale} messages={i18n.messages}>
```

Comment:
i don't see where this `LinguiProvider` came from? it's not in the `src/i18n/i18n.ts`

Reply:
@thekip, it's the same one that's already in the repo.

`i18n.messages` is a getter that returns the messages for the **current** locale. To make it work correctly with the implementation of the provider that's in the repo, a private property with all messages should be passed:
```suggestion
    <LinguiProvider locale={i18n.locale} messages={i18n._messages}>
```